cmake/modules/FoundationSwiftSupport.cmake

```diff
@@ -0,0 +1,54 @@
+# Returns the os name in a variable
+#
+# Usage:
+#   get_swift_host_os(result_var_name)
+#
+#
+# Sets ${result_var_name} with the converted OS name derived from
+# CMAKE_SYSTEM_NAME.
+function(get_swift_host_os result_var_name)
```

Comment:
A function to set the value of a variable to the value of another variable seems a bit excessive.
It looks like this is only getting used below to set `swift_os`. I think you can safely replace the uses of `swift_os` in `_foundation_install_target` with `${SWIFT_SYSTEM_NAME}`.